test/unittest/tests/wrappers/mock_session.cpp

```diff
@@ -25,8 +25,12 @@ MockSession::MockSession(WT_SESSION_IMPL *session, std::shared_ptr<MockConnectio
 
 MockSession::~MockSession()
 {
+    WT_CONNECTION_IMPL *connection_impl = _mockConnection->getWtConnectionImpl();
     if (_sessionImpl->block_manager != nullptr)
         __wt_free(nullptr, _sessionImpl->block_manager);
+    if (connection_impl->file_system != nullptr)
+        utils::throwIfNonZero(connection_impl->file_system->terminate(
```

Comment:
Can we add a comment here explaining why we need to do this / what it does? I don't fully understand why destroying a session would require we terminate something using the connection.

Reply:
You can give it a nullptr in place of the session and use the connection to terminate it.

Reply:
As discussed created ticket [WT-13505](https://jira.mongodb.org/browse/WT-13505)

Reply:
Needs a fix me too.